### DISCUSSION TITLE
Is there any way to specify an age tolerance for matching

### DISCUSSION THREAD

**Discussion by M-Hsiao** [moj-analytical-services/splink#2158] [Q&A]
Hello, I am working with Splink to match individuals across different census iterations, and I found it hard to specify a tolerance range for age comparison. For example:

<!DOCTYPE html><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; caret-color: rgb(31, 35, 40); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">Table  1</p>
ID | first_name | last_name | age
-- | -- | -- | --
1 | John | Smith | 21

<br class="Apple-interchange-newline">

<!DOCTYPE html><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; caret-color: rgb(31, 35, 40); color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">Table 2</p>
ID | first_name | last_name | age
-- | -- | -- | --
2 | Jhn | Smith | 33

<br class="Apple-interchange-newline">

Though, there is a typo error in the first name, according to the surname and age (age difference ≈ census interval), these two records should belong to the same person.

I have tried to specify this in comparison level with a SQL expression "sql_condition": "abs(r.age - l.age) > 5 AND abs(r.age - l.age) <= 15", but didn't work.

Could you help me to figure this out? Thanks a million in advance!

**Comment by RobinL** [comment]
When you say 'it didn't work', what happened?  Did you get an error?  The sql condition looks reasonable to me, so long as age is a numeric column not a text column
